### ISSUE TITLE
[jit] RuntimeError : Wrong type for attribute assignment. Expected Tensor but got Tensor.

### ISSUE THREAD

**Issue by wizardk** [pytorch/pytorch#25860]
## 🐛 Bug

Causing runtime error in code of reassigning a member tensor when calling torch.jit.script

## To Reproduce

```
class MyModule(torch.nn.Module):
    def __init__(self):
        super(MyModule, self).__init__()
        self.v = torch.randn(1)

    def update_v(self):
        self.v = torch.randn(1)

    def forward(self):
        self.update_v()  # <--- HERE

model = torch.jit.script(MyModule())
```

## Expected behavior

No error.

## Environment

 - PyTorch Version : 1.2.0
 - OS : Windows
 - How you installed PyTorch : conda
 - Python version: 3.6.6

cc @suo

**Comment by suo** [comment]
Hi, this is probably because we are refining the type of the attribute `v` (since it's a statically known attribute) but we are not refining the type in `update_v()`

Either way: can you try on master? We have made changes to Tensor typing that should fix this issue

**Comment by wizardk** [comment]
Hi @suo , sorry, I didn't test on master or nightly version yet.
Currently I'm using copy_ to reassign member tensor.

I'll test this case when I'm free and I'll let you know the result.
Thanks for your help.

**Comment by simonhollis** [comment]
Hi @suo, has there been any more progress on this issue. I'm seeing a similar case where calling `torch.jit.script()` seems to be refining a binary member variable to a `Tensor` instance, causing an `if` statement to crash out. Might the issues be related?

**Comment by driazati** [comment]
This issue should be fixed with #24284, can you post a repro of your error @simonhollis?

**Comment by suo** [comment]
Going to close this out as there hasn't been much activity. @simonhollis feel free to reopen if you're still seeing this behavior.
